integration/tests/common/approvalstate/result_approval_state.go

```diff
@@ -12,43 +12,46 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
-const resultApprovalTimeout = 100 * time.Second
+const resultApprovalTimeout = 120 * time.Second
 
 // ResultApprovalState keeps track of the result approval messages getting disseminated through
 // test net network.
 type ResultApprovalState struct {
-	resultApprovals sync.Map
+	sync.RWMutex
+	resultApprovals map[flow.Identifier][]*flow.ResultApproval
+}
+
```

Comment:
No new logic added, just encapsulation in a separate package + making access threadsafe.